pkg/gardenadm/botanist/kubelet.go

```diff
@@ -119,3 +123,40 @@ func (b *AutonomousBotanist) BootstrapKubelet(ctx context.Context) error {
 
 	return b.DBus.Restart(ctx, nil, nil, kubelet.UnitName)
 }
+
+// ApproveKubeletServerCertificateSigningRequest approves the kubelet server certificate signing request.
+func (b *AutonomousBotanist) ApproveKubeletServerCertificateSigningRequest(ctx context.Context) error {
+	serverCertificateExists, err := b.FS.Exists(filepath.Join(kubelet.PathKubeletDirectory, "pki", "kubelet-server-current.pem"))
+	if err != nil {
+		return fmt.Errorf("failed checking if kubelet server certificate exists: %w", err)
+	}
+	if serverCertificateExists {
+		return nil
+	}
+
+	csrList := &certificatesv1.CertificateSigningRequestList{}
+	if err := b.SeedClientSet.Client().List(ctx, csrList); err != nil {
+		return fmt.Errorf("failed listing certificate signing requests: %w", err)
+	}
+
+	for _, csr := range csrList.Items {
+		if csr.Spec.Username == "system:node:"+b.HostName {
+			if !slices.ContainsFunc(csr.Status.Conditions, func(condition certificatesv1.CertificateSigningRequestCondition) bool {
+				return condition.Type == certificatesv1.CertificateApproved && condition.Status == corev1.ConditionTrue
+			}) {
+				b.Logger.Info("Approving kubelet server certificate signing request")
```

Comment:
Do we want to add context, e.g. name of CSR, or is this so obvious that we do not need it in case something went wrong?